src/java/main/org/apache/zookeeper/server/auth/SASLAuthenticationProvider.java

```diff
@@ -38,11 +38,6 @@ public String getScheme() {
     }
 
     public boolean matches(String id,String aclExpr) {
-        if (System.getProperty("zookeeper.superUser") != null) {
-            if (id.equals(System.getProperty("zookeeper.superUser")) || id.equals(aclExpr)) {
-              return true;
-            }
-        }
         if ((id.equals("super") || id.equals(aclExpr))) {
```

Comment:
Thanks @revans2 for the details.
On second thought, it would not be appropriate to break the backward compatibility. So can you please revert changes done for this comment. Rest all looks good to me.